### PR TITLE
feat: GitHub Pages web demo

### PR DESCRIPTION
**Pull request by simbleau**
This will likely need attention from @raphlinus or someone who has administrative privileges on the repo to setup GitHub Pages.

This PR is aimed to package the `with_winit` demo to WASM and serve it with GitHub Pages!
(As soon as #482 is closed - I put a PR for that)

Tasks:
- [x] Wait for WASM to work - #482 
- [x] Enable environments in Admin settings - Needs @raphlinus or @DJMcNab 
- [x] Approve and Merge this PR
- [ ] Update the repo link to https://linebender.github.io/vello

The end result will look like this, except be available on [https://linebender.github.io/vello](https://linebender.github.io/vello)

image

### CODE REVIEW

File: .github/workflows/web-demo.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+env:
+  # We aim to always test with the latest stable Rust toolchain, however we pin to a specific
+  # version like 1.70. Note that we only specify MAJOR.MINOR and not PATCH so that bugfixes still
+  # come automatically. If the version specified here is no longer the latest stable version,
+  # then please feel free to submit a PR that adjusts it along with the potential clippy fixes.
+  RUST_STABLE_VER: "1.76" # In quotes because otherwise (e.g.) 1.70 would be interpreted as 1.7
```
Comment (DJMcNab) [inline .github/workflows/web-demo.yml:6]: We're not using clippy here, so there's no risk of the workflow breaking between runs. I'd suggest just using `stable`

File: .github/workflows/web-demo.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+name: Web Demo Release
```
Comment (DJMcNab) [inline .github/workflows/web-demo.yml:8]:
```suggestion
name: Web Demo Update
```

File: .github/workflows/web-demo.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                    margin: 0;
+                    padding: 0;
+                  }
+                </style>
+              </body>
+            </html>
+          EOF
+
+      - name: deploy
+        uses: peaceiris/actions-gh-pages@v3
```
Comment (DJMcNab) [inline .github/workflows/web-demo.yml:63]: Why is this not using https://github.com/marketplace/actions/deploy-github-pages-site? (i.e. the official one from GitHub). We at least need a comment here explaining that choice

File: README.md [hunk 4]
```diff
@@ -48,6 +48,8 @@ Examples must be selected using the `--package` (or `-p`) Cargo flag.
 
 ### Winit
 
+This example has a demo [available here](https://linebender.github.io/vello) on supporting web browsers.
```
Comment (DJMcNab) [inline README.md:51]: I'd put this in the "Web" section, to leave it in context that we don't actively support the web, and that it only works on the browsers which support WebGPU

**Review by DJMcNab — COMMENTED** [review]
Thanks!

I think this is probably good to have, but there are a few things I'd prefer to be swapped around slightly

File: .github/workflows/web-demo.yml (hunk 3 above)
Comment (simbleau) [reply .github/workflows/web-demo.yml:56]: Added a comment TODO to revisit when stable. While `actions/deploy-pages` is stable, they require you to use [`actions/upload-pages-artifact`](https://github.com/actions/upload-pages-artifact) which is not stable.

**Comment by simbleau** [comment]
@DJMcNab ready again. :)

File: .github/workflows/web-demo.yml (hunk 3 above)
Comment (DJMcNab) [reply .github/workflows/web-demo.yml:56]: I think I'd still much rather us use the official GitHub one - creating a gh-pages branch is messy, and requires additional tokens to be managed.

Like yes, there's risk that things break, but needing to do secrets securely is something I don't want to deal with at the moment

Comment (simbleau) [reply .github/workflows/web-demo.yml:56]: Do you mean `secrets.GITHUB_TOKEN`? This is a token provided to all jobs by GitHub, not something needing to be added or managed by us.

I can still work on the official one, though. I'll have it up soon.

Comment (DJMcNab) [reply .github/workflows/web-demo.yml:56]: Ah, thanks for the pointer. I'd never dug into it before. My prior experience was with talking to external systems, which of course do need tokens managing.

Even still, I do think it's nicer to not make a gh-pages branch, so thanks for changing that

**Review by DJMcNab — APPROVED** [review]
I'm happy to experiment with this - I can't run the example, but I think we added an error page at some point?

**Comment by simbleau** [comment]
> I'm happy to experiment with this - I can't run the example, but I think we added an error page at some point?

The pipeline has to run first. I just kicked it off. Hope it works first try 🤞

**Pull request merged**